lib/transaction.js

```diff
@@ -163,79 +175,69 @@ class Transaction extends EventEmitter {
     // Wait for the earlier Transactions to complete before proceeding.
     await this._previousSibling;
 
-    return this.acquireConnection(config, (connection) => {
-      const trxClient = (this.trxClient = makeTxClient(
-        this,
-        this.client,
-        connection
-      ));
-      const init = this.client.transacting
-        ? this.savepoint(connection)
-        : this.begin(connection);
-      const executionPromise = new Promise((resolver, rejecter) => {
-        this._resolver = resolver;
-        this._rejecter = rejecter;
-      });
-
-      init
-        .then(() => {
-          return makeTransactor(this, connection, trxClient);
-        })
-        .then((transactor) => {
-          transactor.executionPromise = executionPromise;
-
-          // If we've returned a "thenable" from the transaction container, assume
-          // the rollback and commit are chained to this object's success / failure.
-          // Directly thrown errors are treated as automatic rollbacks.
-          let result;
-          try {
-            result = container(transactor);
-          } catch (err) {
-            result = Promise.reject(err);
-          }
-          if (result && result.then && typeof result.then === 'function') {
-            result
-              .then((val) => {
-                return transactor.commit(val);
-              })
-              .catch((err) => {
-                return transactor.rollback(err);
-              });
-          }
-          return null;
-        })
-        .catch((e) => {
-          return this._rejecter(e);
+    const { withConnection } = config;
+
+    const withEverything = pipe([
+      withConnection,
+      InjectTransactionID(this.txid),
+    ]);
```

Comment:
I'm not sure if I'll keep the `pipe([...])` function in the re-implementation.  It's a functional programming concept that many developers might find mystifying.